File: backend/src/main/java/com/api/health/HealthApp/entities/Insurance.java
package com.api.health.HealthApp.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@Table(name="insurances")
public class Insurance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String country;

    private BigDecimal annualPrice;

    @ElementCollection
    @CollectionTable(
            name = "insurances_hospitals_covered",
            joinColumns = @JoinColumn(name = "insurance_id")
    )
    @Column(name = "hospitals_covered")
    private List<Long> hospitalsCovered = new ArrayList<>();

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: frontend/lib/models/insurance.dart
class Insurance {
  final int? id;
  final String name;
  final String country;
  final List<String> hospitalsCovered;
  final double annualPrice;

  Insurance({
    this.id,
    required this.name,
    required this.country,
    required this.hospitalsCovered,
    required this.annualPrice,
  });

  factory Insurance.fromJson(Map<String, dynamic> json) {
    return Insurance(
      id: json['id'],
      name: json['name'],
      country: json['country'],
      hospitalsCovered: List<String>.from(json['hospitalsCovered']),
      annualPrice: json['annualPrice'].toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'country': country,
      'hospitalsCovered': hospitalsCovered,
      'annualPrice': annualPrice,
    };
  }
}